### ISSUE TITLE
Location not working on release web build

### ISSUE THREAD

**Issue by Inihood1** [Lyokone/flutterlocation#817]
I am not sure what am doing wrong, the code works perfectly on debug mode but does not response at all in release build.
I cannot request permission nor can I get coordinates.

Here is my flutter doctor

[√] Flutter (Channel stable, 3.7.5, on Microsoft Windows [Version 10.0.19044.2486], locale en-GB)
    • Flutter version 3.7.5 on channel stable at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c07f788888 (13 days ago), 2023-02-22 17:52:33 -0600
    • Engine revision 0f359063c4
    • Dart version 2.19.2
    • DevTools version 2.20.1

This is the test website: http://attendance.ccsl.ng/
When click on "request" the browser is suppose to popup a location confirmation dialog which is working perfectly in debug mode.

And this is the command I use for web release build: flutter build web

**Comment by EBBozkurt** [comment]
Same here, debug is fine but in release build nothing happen. I also checked in both open and close issues, however, I did not find anything useful. For now, I will replace the package.

**Comment by Inihood1** [comment]
What package will you be switching to? 
Can you let me know how you intend to solve it?
Can we also connect?

**Comment by EBBozkurt** [comment]
I switched geolocator package instead of location. But here is the deal, your website has no **SSL**, and according to [Secure Contexts ](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) in web development, you can only take locations if you have **Secure connection**.

I got this info after changing the package. So here's my suggestion, first do not change your package and get secure connection (You can use firebase deployment, it has free SSL) and try it again. Second if it is fail then change your package to geolocator. But, probably, it will run, because [this ](https://location.bernos.dev/#/) is the website of the package and it works properly.  On the other hand, geolocator with SSL runs good too. Also you can use example to implement your code.

**Comment by sgehrman** [comment]
same issue.  Why is this ignored?

**Comment by Inihood1** [comment]
> same issue. Why is this ignored?

It is working but your website has to be secured with SSL

**Comment by bldev2473** [comment]
> I switched geolocator package instead of location. But here is the deal, your website has no **SSL**, and according to [Secure Contexts ](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) in web development, you can only take locations if you have **Secure connection**.
> 
> I got this info after changing the package. So here's my suggestion, first do not change your package and get secure connection (You can use firebase deployment, it has free SSL) and try it again. Second if it is fail then change your package to geolocator. But, probably, it will run, because [this ](https://location.bernos.dev/#/) is the website of the package and it works properly. On the other hand, geolocator with SSL runs good too. Also you can use example to implement your code.

@EBBozkurt, Hello there, Do you know is it working on safari browser on iOS? I tested this feature on chrome and safari, but it seems that it doesnt work in safari though I have allowed location in settings.

**Comment by EBBozkurt** [comment]
> > I switched geolocator package instead of location. But here is the deal, your website has no **SSL**, and according to [Secure Contexts ](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) in web development, you can only take locations if you have **Secure connection**.
> > I got this info after changing the package. So here's my suggestion, first do not change your package and get secure connection (You can use firebase deployment, it has free SSL) and try it again. Second if it is fail then change your package to geolocator. But, probably, it will run, because [this ](https://location.bernos.dev/#/) is the website of the package and it works properly. On the other hand, geolocator with SSL runs good too. Also you can use example to implement your code.
> 
> @EBBozkurt, Hello there, Do you know is it working on safari browser on iOS? I tested this feature on chrome and safari, but it seems that it doesnt work in safari though I have allowed location in settings.

Hello, yes I can confirm `geolocator` package works on iOS Safari Browser. We currently have clients who is using iOS Safari browser and filling their address on our app. Unfortunately, I can not confirm `location` package  beacuse as I said my earlier mention, I swapped the packages.

However, when I go to the official web page on my real device (iOS) (https://location.bernos.dev/#/), this package works very well and gives my exact location. So this package works iOS Safari browser too.

Long story short both packages works on iOS safari browser :)

I hope it helps.

**Comment by bldev2473** [comment]
> > > I switched geolocator package instead of location. But here is the deal, your website has no **SSL**, and according to [Secure Contexts ](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) in web development, you can only take locations if you have **Secure connection**.
> > > I got this info after changing the package. So here's my suggestion, first do not change your package and get secure connection (You can use firebase deployment, it has free SSL) and try it again. Second if it is fail then change your package to geolocator. But, probably, it will run, because [this ](https://location.bernos.dev/#/) is the website of the package and it works properly. On the other hand, geolocator with SSL runs good too. Also you can use example to implement your code.
> > 
> > 
> > @EBBozkurt, Hello there, Do you know is it working on safari browser on iOS? I tested this feature on chrome and safari, but it seems that it doesnt work in safari though I have allowed location in settings.
> 
> Hello, yes I can confirm `geolocator` package works on iOS Safari Browser. We currently have clients who is using iOS Safari browser and filling their address on our app. Unfortunately, I can not confirm `location` package beacuse as I said my earlier mention, I swapped the packages.
> 
> However, when I go to the official web page on my real device (iOS) (https://location.bernos.dev/#/), this package works very well and gives my exact location. So this package works iOS Safari browser too.
> 
> Long story short both packages works on iOS safari browser :)
> 
> I hope it helps.

Thank you for your information. It helps. 👍

**Comment by omarjericho31** [comment]
This is still an issue. 
My website uses SSL btw.

**Comment by manpreetN224** [comment]
> I switched geolocator package instead of location. But here is the deal, your website has no **SSL**, and according to [Secure Contexts ](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) in web development, you can only take locations if you have **Secure connection**.
> 
> I got this info after changing the package. So here's my suggestion, first do not change your package and get secure connection (You can use firebase deployment, it has free SSL) and try it again. Second if it is fail then change your package to geolocator. But, probably, it will run, because [this ](https://location.bernos.dev/#/) is the website of the package and it works properly. On the other hand, geolocator with SSL runs good too. Also you can use example to implement your code.

Hi I am getting this CoreLocationProvider: CoreLocation framework reported a kCLErrorLocationUnknown failure.
